src/Data/Food/Query.elm

```diff
@@ -412,3 +360,381 @@ parseBase64Query =
         b64decode
             >> Result.toMaybe
             >> Just
+
+
+
+---- Example recipes
+
+
+type alias Product =
+    { name : String
+    , query : Query
+    , category : String
+    }
+
+
+recipesAndNames : List Product
+recipesAndNames =
+    [ { name = "Produit vide", query = emptyQuery, category = "" }
+    , { name = "Carrot cake (643g)", query = carrotCake, category = "Pâtisserie" }
+    , { name = "Épinards congelés (815g)", query = frozenSpinach, category = "Produit surgelé" }
+    , { name = "Pizza jambon fromage congelée (474g)", query = frozenPizzaHamCheese, category = "Produit surgelé" }
+    , { name = "Pizza margharita congelée (662g)", query = frozenPizzaMargarita, category = "Produit surgelé" }
+    , { name = "Ratatouille en conserve (804g)", query = cannedRatatouille, category = "Conserve" }
+    , { name = "Raviolis en conserve (733g)", query = cannedRaviolis, category = "Conserve" }
+    ]
```

Comment:
I'd naturally expect this to live in a JSON file because we already have the appropriate decoder so that would be basically matter of loading over HTTP on the Food page's `init`… I'm pretty sure that wouldn't be much more code and complexity than what this patch features, though I've been away from the codebase for a few weeks so I may be wrong.

Reply:
Sure, we could, but what good would it do us? What's the drawback of having it as elm data?
Having it as elm data:
+ no cpu used to decode
+ no need for another json loading management
- makes the base bundle slightly larger, even for users that will never use the food simulator
- needs some manual typing the first time the example is added in the codebase (compared to simply copy/pasting the json from the "share" tab in the simulator

At the moment we have textile examples as elm structures in `src/Data/Textile/Inputs.elm`, so I guess for coherence it would mean also converting those to json files loaded on demand?

Reply:
> At the moment we have textile examples as elm structures in src/Data/Textile/Inputs.elm, so I guess for coherence it would mean also converting those to json files loaded on demand?

Yes. In the same spirit, why don't we store processes, products, materials, ingredients and so on as Elm code directly? 

In my humble opinion, storing data in external JSON files kinda officially marks them as external data sources, which helps them identifying them as such and designing tooling for administrating and handling them. Maybe tomorrow we might want an example editor? Maybe we'd like offering people to contribute recipes through the github UI? 

Sure these ideas are not "official" requirements from POs, but it basically costs **nothing** to make it possible with very few constraints… I also won't debate on the "CPU parsing cost" argument as I'm confident it's pretty much negligible, and the "another json loading management" one is basically three lines of code as we're already loading a bunch of them through the codebase and shouldn't be much of a big deal.

I won't die on this hill though so I'll let you decide whatever you eventually want to do with this simple idea.

Reply:
I have a feeling that this will cost a bit more than **nothing** to change, so I'll land this one, and start a new PR straight away to experiment and see where this goes.
Thanks for the feedback